applications/luci-app-throughput-measurements/root/usr/libexec/rpcd/luci.tputmeas

```diff
@@ -0,0 +1,238 @@
+local nixio = require "nixio"
+local fs = require "nixio.fs"
+local json = require "luci.jsonc"
+local sys = require "luci.sys"
+local posix = require "posix"
+
+
+local methods = {
+	iperf3Start = {
+		args = { mode = "mode", server_address = "server_address", interface = "interface",
+				interface_address = "interface_address", port = "port", duration = "duration",
+				bytes = "bytes", udp = "udp",  bitrate = "bitrate", reverse = "reverse",
+				connections = "connections", interval = "interval" },
+		call = function(args)
+			local iperf3_cmd = "iperf3 --forceflush "
+
+			if fs.stat("/var/run/iperf3.pid") then
+				return { result = "error: Iperf3 is running" }
+			else
+				if not args then
+					return { result = "error: Lack of arguments" }
+				end
+
+				if (not args.mode or (args.mode ~= "-c" and args.mode ~= "-s")) then
+					return { result = "error: mode has not been specified " }
+				end
+
+				iperf3_cmd = iperf3_cmd .. args.mode
+
+				if args.server_address and args.server_address ~= '' then
+					local ipparts = {args.server_address:match("^(%d+)%.(%d+)%.(%d+)%.(%d+)$")}
+					if #ipparts == 4 then
+						for _,nb in pairs(ipparts) do
+							if tonumber(nb) > 255 then
+								return { result = "error: Incorrect format of a server address" }
+							end
+						end
+					else
+						return { result = "error: Incorrect format of a server address" }
+					end
+
+					iperf3_cmd = iperf3_cmd .. " " .. args.server_address
+				end
+
+				if  args.interface and args.interface ~= '' then
+					local ifa = nixio.getifaddrs()
+					for k, v in pairs(ifa) do
+						if ifa[k].name == args.interface and ifa[k].family == "inet" and ifa[k].addr then
+							iperf3_cmd = iperf3_cmd .. " -B " .. ifa[k].addr
+						end
+					end
+				elseif args.interface_address and args.interface_address ~= '' then
+					local ifa = nixio.getifaddrs()
+					local ok = false
+					for k, v in pairs(ifa) do
+						if ifa[k].addr == args.interface_address then
+							iperf3_cmd = iperf3_cmd .. " -B " .. ifa[k].addr
+							ok = true
+						end
+					end
+					if not ok then
+						return { result = "error: Incorrect format of an interface address" }
+					end
+				end
+
+				if args.port and args.port ~= '' then
+					if tonumber(args.port) then
+						iperf3_cmd = iperf3_cmd .. " -p " .. args.port
+					else
+						return { result = "error: Incorrect port argument" }
+					end
+				end
+
+				if args.duration and args.duration ~= '' then
+					if tonumber(args.duration) then
+						iperf3_cmd = iperf3_cmd .. " -t " .. args.duration
+					else
+						return { result = "error: Incorrect duration argument" }
+					end
+				end
+
+				if args.bytes and args.bytes ~= '' then
+					if args.bytes:match("^[0-9]*$") or args.bytes:match("^[0-9]*[kKmM]$") then
+						iperf3_cmd = iperf3_cmd .. " -n " .. args.bytes
+					else
+						return { result = "error: Incorrect bytes argument" }
+					end
+				end
+
+				if args.udp == "1" and args.udp ~= '' then
```

Comment:
The checks are redundant. But why not using a boolean? Would simplify this into `if args.udp then`